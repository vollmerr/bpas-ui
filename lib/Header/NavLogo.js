var _templateObject = _taggedTemplateLiteral(['\n  display: inline-block;\n'], ['\n  display: inline-block;\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  margin: ', ';\n  height: 60px;\n'], ['\n  margin: ', ';\n  height: 60px;\n']);

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import logo from './odiLogo.png';

var Wrapper = styled(Link)(_templateObject);

var Image = styled.img(_templateObject2, function (props) {
  return props.isMobile ? '15px 0' : '0 15px 0 0';
});

/** Renders the navbar logo  */
function NavLogo(_ref) {
  var isMobile = _ref.isMobile,
      onClick = _ref.onClick;

  return React.createElement(
    Wrapper,
    { to: '/', onClick: onClick },
    React.createElement(Image, {
      src: logo,
      alt: 'CA Department of Technology logo',
      isMobile: isMobile
    })
  );
}

NavLogo.PropTypes = {
  onClick: PropTypes.func
};

export default NavLogo;