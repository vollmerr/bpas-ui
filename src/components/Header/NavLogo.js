import React from 'react';
import styled from 'styled-components';
import logo from './_odiLogo.png';

const Wrapper = styled.a`
  margin: ${props => props.isMobile ? '15px 0' : '0'};
  display: inline-block;
  height: 60px;
`;

const Image = styled.img`
  height: 60px;
`;

/** Renders the navbar logo  */
function NavLogo(props) {
  return (
    <Wrapper href={'/'} {...props}>
      <Image src={logo} alt={'CA Department of Technology logo'} />
    </Wrapper>
  );
}

export default NavLogo;
