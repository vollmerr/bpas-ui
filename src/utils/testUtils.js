import React from 'react';
import { StaticRouter } from 'react-router';

export function testRouter(component) {
  const context = {};
  const location = './';

  return (
    <StaticRouter location={location} context={context}>
      {component}
    </StaticRouter>
  );
}
