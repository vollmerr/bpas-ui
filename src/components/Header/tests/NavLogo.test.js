import React from 'react';
import renderer from 'react-test-renderer';
import { testRouter } from '../../../utils/testUtils.js';
import NavLogo from '../NavLogo';

describe('<NavLogo />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      testRouter(<NavLogo />)
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  xit('passes the correct props down', () => {});
});
