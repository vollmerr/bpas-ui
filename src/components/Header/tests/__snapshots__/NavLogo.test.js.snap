// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLogo /> renders correctly 1`] = `
<a
  className="sc-bdVaJa bSrzPW"
  href="/"
  onClick={[Function]}
>
  <img
    alt="CA Department of Technology logo"
    className="sc-bwzfXH kYVJeC"
    src="_odiLogo.png"
  />
</a>
`;
